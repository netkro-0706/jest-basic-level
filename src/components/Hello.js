export default function Hello({ user }) {
  return user && user.name ? (
    <h1>Hello! {user.name}</h1>
  ) : (
    <button>Login</button>
  );
}
