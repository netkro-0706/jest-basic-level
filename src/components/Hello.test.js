import { render, screen } from "@testing-library/react";
import Hello from "./Hello";

//스냅샷 및 props 테스트
const user = {
  name: "Tom",
  age: 30,
};

const user2 = {
  age: 20,
};

xtest("snapshot test", () => {
  //test skip

  test("snapshot : name 있음", () => {
    const { container } = render(<Hello user={user} />);
    expect(container).toMatchSnapshot();
  });
  test("snapshot : name 없음", () => {
    const { container } = render(<Hello />);
    expect(container).toMatchSnapshot();
  });

  test("Hello글자가 포함 되는가?", () => {
    render(<Hello user={user} />);
    const helloEl = screen.getByText(/Hello/i);
    expect(helloEl).toBeInTheDocument();
  });
});
