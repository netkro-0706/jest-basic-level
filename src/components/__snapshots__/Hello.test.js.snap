// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot : name 없음 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;

exports[`snapshot : name 있음 1`] = `
<div>
  <h1>
    Hello! 
    Tom
  </h1>
</div>
`;
